packages/loader/container-loader/src/container.ts

```diff
@@ -711,7 +712,10 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
 
         assert(this.connectionState === ConnectionState.Disconnected, 0x0cf /* "disconnect event was not raised!" */);
 
-        this.service?.dispose();
+        // Notify storage about critical errors. They may be due to disconnect between client & server knowlege about
+        // file, like file being overwritten in storage, but client having stale local cache.
+        // Driver need to ensure all caches are cleared on critical errors
+        this.service?.dispose(error?.errorType === DriverErrorType.epochVersionMismatch);
```

Comment:
i don't love this abstraction, which error are criticial, and to what drivers, can this be expressed more explicitly. if we follow dipose, we could just give this dispose the error, and let it do what it wants with things like error type

Reply:
That's correct, though I'd use "loader concept" terminology, as runtime for me is container runtime and downstream.
I do not see an easy way out here if we the check for is in deltamanager and we want to not have generic "something wrong happened with a file" messsage.
We could use two different error codes (one from ODSP and one from loader), but I'm not sure it gives us anything.
That said, I think using "epoch" terminology is wrong here, I'll change it to more generic "file changed" kind of terminology.

Reply:
Though it has to be any, not Error, as our errors do not derive (at least types) from Error. We likely can fix it, but it's much bigger change, so I'd prefer to have it in separate PR

Reply:
yeah. that fine, but the shape is closer so if we improve it hopefully back compat is a bit easier